webapp/src/pages/eventConfig/components/EventConfigComponent.js

```diff
@@ -1,19 +1,47 @@
 import React, { Component } from "react";
 import { eventService } from "../../../services/events";
 import { withRouter } from "react-router";
-import DateTimePicker from "react-datetime-picker";
 import { withTranslation } from 'react-i18next';
+import FormTextBox from "../../../components/form/FormTextBox";
+import FormTextArea from "../../../components/form/FormTextArea";
+import FormDate from "../../../components/form/FormDate";
 
 class EventConfigComponent extends Component {
   constructor(props) {
     super(props);
 
+    let today = new Date().toLocaleDateString();
+
+    this.emptyEvent = {
+      name: "",
+      description: "",
+      start_date: today,
+      end_date: today,
+      key: null,
+      organisation_id: this.props.organisation.id,
+      email_from: "",
+      url: "",
+      application_open: today,
+      application_close: today,
+      review_open: today,
+      review_close: today,
+      selection_open: today,
+      selection_close: today,
+      offer_open: today,
+      offer_close: today,
+      registration_open: today,
+      registration_close: today  
+    }
+
     this.state = {
       preEvent: this.emptyEvent,
```

Comment:
I am not quite sure what the preEvent is doing currently. Might it be useful if a user is editing an existing, rather than creating a new, event? Should this form be used for both? If so, I was thinking of adding some conditional displays (e.g. use Update Event or Create Event buttons depending on which state they're in)

Reply:
Yes definitely this form should be used for both editing and creating new events, but even in the editing case I can't really see a use for preEvent, unless maybe you want to do the "field edited" thing for validation :) If using editing mode you would know from the route that you're editing and could just put a simple flag in your state to control the submit button text

Reply:
Got it. Though currently hasBeenEdited is cycling through this.state.preEvent to see if any fields have been edited before updating the state. What is the best way to replace this?